python/dask_cudf/dask_cudf/sorting.py

```diff
@@ -152,24 +152,23 @@ def rearrange_by_hash(
     groups = []
     splits = []
     combines = []
+    inputs = []
 
-    inputs = [
-        tuple(digit(i, j, k) for j in range(stages))
-        for i in range(k ** stages)
-    ]
+    start = {}
+    end = {}
 
     token = tokenize(df, columns, max_branch)
     shuffle_combine_token = "shuffle-combine-" + token
     shuffle_token = "shuffle-" + token
 
-    start = {}
-    end = {}
+    for index, i in enumerate(range(k ** stages)):
```

Comment:
We shouldn't need to enumerate a range, it's already 0 --> N. Could just do `i in range(k ** stages)`.